.github/workflows/build.yml

```diff
@@ -0,0 +1,321 @@
+name: build
+
+on:
+  push:
+    branches:
+    - master
+    tags:
+    - v*
+  pull_request:
+    branches:
+    - master
+  schedule:
+    - cron:  '23 1 * * 0'
+  release:
+    types: [published]
+  workflow_dispatch:
+
+defaults:
+  run:
+    shell: bash
+
+concurrency:
+  group: build-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+
+  cmake-centos7:
+
+    runs-on: ubuntu-latest
+    container: gitlab-registry.cern.ch/linuxsupport/cc7-base:latest
+
+    steps:
+    - name: Install external dependencies with yum
```

Comment:
I'll note that these jobs fail sometimes infrequently because a mirror fails to be found in time. If these get restarted they pass.